osquery/core/posix/utils.cpp

```diff
@@ -41,4 +41,33 @@ Status platformStrncpy(char* dst, size_t nelms, const char* src, size_t count) {
   ::strncpy(dst, src, count);
   return Status(0, "OK");
 }
+
+char* canonicalize_file_name(char* name) {
```

Comment:
May it is a stupid question, but why we are not able to use `std::string` (or even `boost::filesystem::path`) here instead of `char*`?

Reply:
https://www.boost.org/doc/libs/1_55_0/libs/filesystem/doc/reference.html#canonical ?